File: Java/BinaryTreeInorderTraversal/BinaryTreeInorderTraversal.java
import com.phoenix.Util.TreeNode;

import java.util.ArrayList;
import java.util.List;

/**
 * (1ms)
 * Created by Phoenix-Z on 2017/4/21.
 */
public class BinaryTreeInorderTraversal {

    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> ans = new ArrayList<>();
        helper(root, ans);
        return ans;
    }

    private void helper(TreeNode root, List<Integer> ans) {
        if(root == null)
            return;
        helper(root.left, ans);
        ans.add(root.val);
        helper(root.right, ans);
    }

}

File: JavaScript/BinaryTreeInorderTraversal/BinaryTreeInorderTraversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * (112ms)
 * @param {TreeNode} root
 * @return {number[]}
 */
var inorderTraversal = function(root) {
    var ans = [];
    var stack = [];
    var cur = root;
    while(cur !== null || stack.length > 0) {
        while(cur !== null) {
            stack.push(cur);
            cur = cur.left;
        }
        cur = stack.pop();
        ans.push(cur.val);
        cur = cur.right;
    }
    return ans;
};